artifactory/services/utils/artifactoryutils.go

```diff
@@ -232,7 +232,20 @@ func getLatestBuildNumberFromArtifactory(buildName, buildNumber, projectKey stri
 		buildRepo = projectKey
 	}
 	buildRepo += buildRepositoriesSuffix
-	aqlBody := CreateAqlQueryForLatestCreated(buildRepo, buildName)
+	buildNameTrimDelimiter := strings.TrimSuffix(buildName, "/")    // "buildname/" >> "buildname" (remove trailing '/' if exist and there is no build name)
+	buildNameQueryEscape := url.QueryEscape(buildNameTrimDelimiter) // "build/name" >> "build%2Fname" (Escape special characters with url encoding escaping)
```

Comment:
Let's put the above two description comments above the code lines they describe. This makes reading the code easier.

Also, why is this URL encoding needed?

Reply:
The encoding is there to turn '/' to '%2F' (and other special characters) because that is the way they appear in the DB.